Add merchant session and merchant purchase (#131)
starkbank/sdk-python: README.md

    - [CorporateBalance](#get-your-corporatebalance): View your corporate balance
    - [CorporateTransactions](#query-corporatetransactions): View the transactions that have affected your corporate balance
    - [CorporateEnums](#corporate-enums): Query enums related to the corporate purchases, such as merchant categories, countries and card purchase methods
    - [MerchantCard](#query-merchantcard): The Merchant Card resource stores information about cards used in approved purchases. These cards can be used in new purchases without the need to create a new session.
    - [MerchantSession](#create-a-merchantsession): The Merchant Session allows you to create a session prior to a purchase. Sessions are essential for defining the parameters of a purchase, including funding type, expiration, 3DS, and more.
    - [MerchantPurchase](#create-a-merchantpurchase): The Merchant Purchase resource can be used to charge customers with credit or debit cards. If a card hasn't been used before, a Merchant Session Purchase must be created and approved with that specific card before it can be used directly in a Merchant Purchase.

>     - [MerchantInstallment](#query-merchantinstallment): Merchant Installments are created for every installment in a purchase. These resources will track its own due payment date and settlement lifecycle.

Reviewer: Deixa essas descrições mais breves, que nem os outros lugares.
Exemplo pra MerchantSession: "Manage a session to run a purchase"

    - [CorporateBalance](#get-your-corporatebalance): View your corporate balance
    - [CorporateTransactions](#query-corporatetransactions): View the transactions that have affected your corporate balance
    - [CorporateEnums](#corporate-enums): Query enums related to the corporate purchases, such as merchant categories, countries and card purchase methods

>     - [MerchantSession](#merchant-session): The Merchant Session allows you to create a session prior to a purchase. Sessions are essential for defining the parameters of a purchase, including funding type, expiration, 3DS, and more.

Reviewer: ```suggestion
    - [MerchantSession](#create-a-merchantsession): Manage a session to run a purchase
```

>     - [MerchantPurchase](#merchant-purchase): The Merchant Purchase section allows users to retrieve detailed information of the purchases.

Reviewer 2: Adicionar info de que é possivel criar uma purchase pelo resource merchant-purchase, não apenas consultá-la. Da forma que está, me passa a ideia de que a purchase é criada apenas pela session e que o resource merchant-purchase serve apenas para consulta.

>     - [MerchantPurchase](#merchant-purchase): The Merchant Purchase resource can be used to charge customers with credit or debit cards. If a card hasn't been used before, a Merchant Session Purchase must be created and approved with that specific card before it can be used directly in a Merchant Purchase.

Reviewer: ```suggestion
    - [MerchantPurchase](#create-a-merchantpurchase): Charge customers with credit or debit cards.
```

Reviewer: Não faltou documentar MerchantCard e MerchantInstallment?

print(log)
```
## Merchant Session

The Merchant Session allows you to create a session prior to a purchase.
Sessions are essential for defining the parameters of a purchase, including funding type, expiration, 3DS, and more.

## Create a MerchantSession

```python
import starkbank

merchant_session = starkbank.merchantsession.create({
    "allowedFundingTypes": [
        "debit",
        "credit"
    ],
    "allowedInstallments": [
        {
            "totalAmount": 0,
            "count": 1
        },
        {
            "totalAmount": 120,
            "count": 2
        },
        {
            "totalAmount": 180,
            "count": 12
        }
    ],
    "expiration": 3600,
    "challengeMode": "disabled",
    "tags": [
        "yourTags"
    ]
})

print(merchant_session)
```

You can create a MerchantPurchase through a MerchantSession by passing its UUID.
**Note**: This method must be implemented in your front-end to ensure that sensitive card data does not pass through the back-end of the integration.

### Create a MerchantSession Purchase

```python
import starkbank

> merchant_session_purchase = starkbank.merchantsession.purchase(

Reviewer 2: mudar os parâmetros para algo mais proximo do que é utilizado nos outros produtos. Exemplo: holder_name="Tywin Lannister", holder_email="tywinLannister@casterlyrock.com", billing_street_line1="Casterly Rock, 1000", etc

  uuid="0bb894a2697d41d99fe02cad2c00c9bc",
  amount=180,
  installment_count=12,
  card_expiration="2035-01",
  card_number="5448280000000007",
  card_security_code="123",
  holder_name="Margaery Tyrell",
  holder_email="margaery.tyrell@email.com",
  holder_phone="11998663456",
  funding_type="credit",
  billing_country_code="BRA",
  billing_city="São Paulo",
  billing_state_code="SP",
  billing_street_line1="Rua do Jardim de cima, 123",
  billing_street_line2="1 andar",
  billing_zip_code="11111-111",
  metadata={
    "extraData": "extraData",
    "language": "pt-BR",
    "timezoneOffset": 3,
    "userAgent": "Mozilla",
    "userIp": "255.255.255.255"
  }
)

print(merchant_session_purchase)
```
### Query MerchantSessions
```python
import starkbank

merchant_sessions = starkbank.merchantsession.query(limit=3)
for merchant_session in merchant_sessions:
    print(merchant_session)
```
### Get a MerchantSession
```python
import starkbank

merchant_session = starkbank.merchantsession.get('5950134772826112')
print(merchant_session)
```
## Merchant Purchase

> The Merchant Purchase section allows users to retrieve detailed information of the purchases.

Reviewer 2: Mesmo ponto: como, via sdk, o cara criaria uma merchant-purchase a partir de um card id?

print(log)
```

> ## Merchant Session

Reviewer: Deixa o espaçamento de 1 enter entre cada um dos recursos.
Faz isso em todos outros lugares que você codou

> ## Merchant Session
> 
> The Merchant Session allows you to create a session prior to a purchase.
> Sessions are essential for defining the parameters of a purchase, including funding type, expiration, 3DS, and more.
> 
> ## Create a MerchantSession
> 

Reviewer: ```suggestion
## Create a MerchantSession

The Merchant Session allows you to create a session prior to a purchase.
Sessions are essential for defining the parameters of a purchase, including funding type, expiration, 3DS, and more.

```

```python
import starkbank

merchant_session = starkbank.merchantsession.create({
    "allowedFundingTypes": [
        "debit",
        "credit"
    ],
    "allowedInstallments": [
        {
            "totalAmount": 0,
            "count": 1
        },
        {
            "totalAmount": 120,
            "count": 2
        },
        {
            "totalAmount": 180,
            "count": 12
        }
    ],
    "expiration": 3600,
    "challengeMode": "disabled",
    "tags": [

>         "yourTags"

Reviewer: As tags no sistema sempre são transformadas pra lower case.
Nesse caso, eu colocaria a tag de exemplo em kebab-case.

    ]
})

print(merchant_session)
```

> You can create a MerchantPurchase through a MerchantSession by passing its UUID.
> **Note**: This method must be implemented in your front-end to ensure that sensitive card data does not pass through the back-end of the integration.
> 
> ### Create a MerchantSession Purchase
> 

Reviewer: ```suggestion
**Note**: This method must be implemented in your front-end to ensure that sensitive card data does not pass through the back-end of the integration.

## Create a MerchantSession Purchase

You can create a MerchantPurchase through a MerchantSession by passing its UUID.

```

```python
import starkbank

> merchant_session_purchase = starkbank.merchantsession.purchase(

Reviewer: Coloca a indentação no padrão de 4 espaços.

  uuid="0bb894a2697d41d99fe02cad2c00c9bc",

>   amount=180,

Reviewer: Melhor aumentar o amount dessa compra, R$1,80 parcelado em 12 vezes é diferenciado hahaha.

  installment_count=12,
  card_expiration="2035-01",
  card_number="5448280000000007",
  card_security_code="123",
  holder_name="Margaery Tyrell",
  holder_email="margaery.tyrell@email.com",
  holder_phone="11998663456",
  funding_type="credit",
  billing_country_code="BRA",
  billing_city="São Paulo",
  billing_state_code="SP",
  billing_street_line1="Rua do Jardim de cima, 123",
  billing_street_line2="1 andar",
  billing_zip_code="11111-111",
  metadata={
    "extraData": "extraData",
    "language": "pt-BR",
    "timezoneOffset": 3,
    "userAgent": "Mozilla",
    "userIp": "255.255.255.255"
  }
)

print(merchant_session_purchase)
```

> ### Query MerchantSessions

Reviewer: Troca esses headers (###) pra ##

Reviewer: Faz isso pros outros casos que você codou

```python
import starkbank

merchant_sessions = starkbank.merchantsession.query(limit=3)
for merchant_session in merchant_sessions:
    print(merchant_session)
```
### Get a MerchantSession
```python
import starkbank

merchant_session = starkbank.merchantsession.get('5950134772826112')
print(merchant_session)
```

> ## Merchant Purchase
> The Merchant Purchase resource can be used to charge customers with credit or debit cards.
> If a card hasn't been used before, a Merchant Session Purchase must be created and approved with that specific card before it can be used directly in a Merchant Purchase.
> 
> ### Create a MerchantPurchases

Reviewer: ```suggestion

### Create a MerchantPurchases

The Merchant Purchase resource can be used to charge customers with credit or debit cards.
If a card hasn't been used before, a Merchant Session Purchase must be created and approved with that specific card before it can be used directly in a Merchant Purchase.

```

```python
import starkbank

> merchant_purchase = starkbank.merchantpurchase.create(

Reviewer: Arruma a indentação pra 4 espaços.

print(log)
```

> ## Query MerchantCard

Reviewer: Altera o anchor link no index também
```suggestion
## Query MerchantCards
```

        funding_type="credit",
        challenge_mode="disabled",
        billing_city="Sao Paulo",
        billing_country_code="BRA",
        billing_state_code="SP",
        billing_street_line_1="Rua Casterly Rock, 2000",
        billing_street_line_2="1 andar",
        billing_zip_code="11111-111",
        holder_email="tywin.lannister@gmail.com",
        holder_phone="11985923451",
        metadata={
            "userAgent": "userAgent",
            "userIp": "255.255.255.255",
            "language": "pt-BR",
            "timezoneOffset": 3,
            "extraData": "extraData"
        },
        tags=["teste"]
    )
)
```

## Query MerchantPurchases

Get a list of merchant purchases in chunks of at most 100. If you need smaller chunks, use the limit parameter.

```python
import starkbank

merchant_purchases = starkbank.merchantpurchase.query(limit=3)
for merchant_purchase in merchant_purchases:
    print(merchant_purchase)
```

## Get a MerchantPurchase

Retrieve detailed information about a specific purchase by its id.

```python
import starkbank

merchant_purchase = starkbank.merchantpurchase.get('5950134772826112')
print(merchant_purchase)
```

> ## Query MerchantInstallment

Reviewer: Altera o anchor link no index também
```suggestion
## Query MerchantInstallments
```
